test/gateway/value-types.js

```diff
@@ -0,0 +1,410 @@
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    url: '/graphql',
+    body: JSON.stringify({
+      query: commentsQuery
+    })
+  })
+
+  t.same(JSON.parse(commentsRes.body), commentsData)
+})
+
+test('Should be able to query with value types and polling', async (t) => {
+  const clock = FakeTimers.install({
+    shouldAdvanceTime: true,
+    advanceTimeDelta: 40
+  })
+  t.teardown(() => clock.uninstall())
+
+  const [userService, userServicePort] = await createService(t, userSchema, userResolvers)
+  const [postService, postServicePort] = await createService(t, postSchema, postResolvers)
+  const [commentService, commentServicePort] = await createService(t, commentSchema, commentResolvers)
+
+  const gateway = Fastify()
+
+  t.teardown(async () => {
+    await gateway.close()
+    await userService.close()
+    await postService.close()
+    await commentService.close()
+  })
+
+  gateway.register(GQL, {
+    gateway: {
+      services: [{
+        name: 'user',
+        url: `http://localhost:${userServicePort}/graphql`
+      }, {
+        name: 'post',
+        url: `http://localhost:${postServicePort}/graphql`
+      }, {
+        name: 'comment',
+        url: `http://localhost:${commentServicePort}/graphql`
+      }],
+      pollingInterval: 2000
+    }
+  })
+
+  const postsRes = await gateway.inject({
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    url: '/graphql',
+    body: JSON.stringify({
+      query: postsQuery
+    })
+  })
+
+  t.same(JSON.parse(postsRes.body), postsData)
+
+  const helloQuery = `
+    query {
+      hello
+    }
+  `
+
+  const helloRes = await gateway.inject({
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    url: '/graphql',
+    body: JSON.stringify({
+      query: helloQuery
+    })
+  })
+
+  t.same(JSON.parse(helloRes.body), {
+    errors: [
+      {
+        message:
+          'Cannot query field "hello" on type "Query".',
+        locations: [{ line: 3, column: 7 }]
+      }
+    ],
+    data: null
+  })
+
+  postService.graphql.replaceSchema(
+    buildFederationSchema(`
+      ${postSchema}
+      extend type Query {
+        hello: String!
+      }
+    `)
+  )
+  postService.graphql.defineResolvers({
+    ...postResolvers,
+    Query: {
+      hello: () => 'world',
+      ...postResolvers.Query
+    }
+  })
+
+  await clock.tickAsync(2000)
+
+  // We need the event loop to actually spin twice to
+  // be able to propagate the change
+  await immediate()
+  await immediate()
+
+  const postsRes2 = await gateway.inject({
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    url: '/graphql',
+    body: JSON.stringify({
+      query: helloQuery
+    })
+  })
+
+  t.same(JSON.parse(postsRes2.body), {
+    data: {
+      hello: 'world'
+    }
+  })
+})
+
+test('Should not register value types if field ', async (t) => {})
```

Comment:
can you fill this in?

Reply:
This seems to be a leftover. Or do you miss a test?

Reply:
looks like it is, just drop it!